src/tests/integration_test_utils.rs

```diff
@@ -152,65 +183,89 @@ impl DeltaTest {
         self
     }
 
-    pub fn with_config_and_input(config: &config::Config, input: &str) -> DeltaTestOutput {
-        DeltaTestOutput {
-            output: run_delta(input, &config),
-            explain_ansi_: false,
-        }
+    pub fn explain_ansi(mut self) -> Self {
+        self.explain_ansi_ = true;
+        self
     }
 
     pub fn with_input(&self, input: &str) -> DeltaTestOutput {
         let _args = FakeParentArgs::for_scope(self.calling_process.as_deref().unwrap_or(""));
-        DeltaTest::with_config_and_input(&self.config, input)
+        let raw = run_delta(input, &self.config);
+        let cooked = if self.explain_ansi_ {
+            ansi::explain_ansi(&raw, false)
+        } else {
+            ansi::strip_ansi_codes(&raw)
+        };
+
+        DeltaTestOutput {
+            raw_output: raw,
+            output: cooked,
+        }
     }
 }
 
 pub struct DeltaTestOutput {
-    output: String,
-    explain_ansi_: bool,
+    pub raw_output: String,
+    pub output: String,
 }
 
 impl DeltaTestOutput {
     /// Print output, either without ANSI escape sequences or, if explain_ansi() has been called,
     /// with ASCII explanation of ANSI escape sequences.
     #[allow(unused)]
     pub fn inspect(self) -> Self {
-        eprintln!("{}", "▼".repeat(100));
-        eprintln!("{}", self.format_output());
-        eprintln!("{}", "▲".repeat(100));
+        eprintln!("{}", delineated_string(&self.output.as_str()));
         self
     }
 
     /// Print raw output, with any ANSI escape sequences.
     #[allow(unused)]
     pub fn inspect_raw(self) -> Self {
-        eprintln!("{}", "▼".repeat(100));
-        eprintln!("{}", self.output);
-        eprintln!("{}", "▲".repeat(100));
+        eprintln!("{}", delineated_string(&self.raw_output));
         self
     }
 
-    pub fn explain_ansi(mut self) -> Self {
-        self.explain_ansi_ = true;
+    pub fn expect_skip(self, skip: usize, expected: &str) -> Self {
+        lines_match_skip(skip, expected, &self.output);
         self
     }
 
-    pub fn expect_skip(self, skip: usize, expected: &str) -> String {
-        let processed = self.format_output();
-        lines_match(expected, &processed, Some(skip));
-        processed
+    pub fn expect(self, expected: &str) -> Self {
+        self.expect_skip(0, expected)
     }
 
-    pub fn expect(self, expected: &str) -> String {
+    pub fn expect_after_hdr(self, expected: &str) -> Self {
```

Comment:
Do you mind if we call this `expect_after_header()`? (I'm not keen on abbreviations in public APIs)

Reply:
Sounds fine to me.  I had even wondered if you might prefer that unabbreviated.